crypto/src/hash/mod.rs

```diff
@@ -0,0 +1,30 @@
+use winterfell::crypto::hashers::Rp64_256 as Hasher;
+use winterfell::crypto::ElementHasher;
+use winterfell::crypto::Hasher as HashFn;
+use winterfell::math::StarkField;
+
+mod rpo;
+pub use rpo::Rpo;
+
+// TYPE ALIASES
+// ================================================================================================
+
+pub type Digest = <Hasher as HashFn>::Digest;
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+#[inline(always)]
+fn exp_acc<B: StarkField, const N: usize, const M: usize>(base: [B; N], tail: [B; N]) -> [B; N] {
+    let mut result = base;
+    for _ in 0..M {
+        result.iter_mut().for_each(|r| *r = r.square());
+    }
+    result.iter_mut().zip(tail).for_each(|(r, t)| *r *= t);
+    result
+}
```

Comment:
Seems like this function is not used anywhere. Do we need to have it here?

Reply:
ok - let's keep it. But in general, the preferred way would be to introduce it in the relevant PR.